FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -0,0 +1,291 @@
+                                   window.windowLevel = +CGFLOAT_MAX;
+
+                                   // Run it.
+                                   [window makeKeyAndVisible];
+
+                                   // performs authentication request
+                                   [FIRAppDistributionAppDelegatorInterceptor sharedInstance]
+                                       .currentAuthorizationFlow = [OIDAuthState
+                                       authStateByPresentingAuthorizationRequest:request
+                                                        presentingViewController:
+                                                            self.safariHostingViewController
+                                                                        callback:^(
+                                                                            OIDAuthState
+                                                                                *_Nullable authState,
+                                                                            NSError
+                                                                                *_Nullable error) {
+                                                                          self.authState =
+                                                                              authState;
+                                                                          self->_isTesterSignedIn =
+                                                                              self.authState ? YES
+                                                                                             : NO;
+
+                                                                          completion(error);
+                                                                        }];
+                                 }];
+}
+
+- (void)signOutTester {
+  self.authState = nil;
+  _isTesterSignedIn = false;
+}
+
+- (void)fetchReleases:(FIRAppDistributionUpdateCheckCompletion)completion {
+  NSLog(@"Token: %@", self.authState.lastTokenResponse.accessToken);
+  NSURLSession *URLSession = [NSURLSession sharedSession];
+  NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+  NSString *urlString =
+      [NSString stringWithFormat:ReleasesEndpointURL, [[FIRApp defaultApp] options].googleAppID];
+  [request setURL:[NSURL URLWithString:urlString]];
+  [request setHTTPMethod:@"GET"];
+  [request setValue:[NSString
+                        stringWithFormat:@"Bearer %@", self.authState.lastTokenResponse.accessToken]
+      forHTTPHeaderField:@"Authorization"];
+
+  NSURLSessionDataTask *listReleasesDataTask =
+      [URLSession dataTaskWithRequest:request
+                    completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                      if (error) {
+                        // TODO: Reformat error into error code
+                        completion(nil, error);
```

Comment:
Is there a reason why in the non-error case, the completion is executed in the main thread but not here? Is this on purpose?

Reply:
This needs to be fixed. We need to reformat the error into an AppDistributionError.

Reply:
Will follow up with that fix in a different PR though